### ISSUE TITLE
Validation loss always lower than fitting loss?

### ISSUE THREAD

**Issue by rdrighetto** [MLI-lab/DeepDeWedge#7]
Hi Simon,

I am consistently observing the validation loss almost always being lower than the fitting loss, which is quite puzzling. Isn't the validation loss expected to be higher than the fitting loss, on average?
My guess is some issue with the normalization of the validation subtomograms, not sure. The curve can probably still be used to identify overfitting, regardless of the scale, I think.

Here are 4 examples from different cases (2 sets of tomograms reconstructed in 2 different ways each):
![metrics_aretomo](https://github.com/MLI-lab/DeepDeWedge/assets/7696255/cab5c70d-fd0f-472c-bea7-0bda6e1f9706)
![metrics_imod](https://github.com/MLI-lab/DeepDeWedge/assets/7696255/19b9c5cd-faf8-4064-92bf-b2ab4b7e198b)
![model_1tomo_box80_chans32_batch5_loss](https://github.com/MLI-lab/DeepDeWedge/assets/7696255/f3154899-3263-476e-867b-94b4005ce61d)
![model_1tomo_box80_chans32_batch5_NEW_loss](https://github.com/MLI-lab/DeepDeWedge/assets/7696255/b41684d0-58c7-42a1-8bd4-23e716a04a12)

Thank you!

**Comment by SimWdm** [comment]
Hi Ricardo,

thank you for sharing this very interesting observation! Here are my thoughts:

The magnitude of the fitting and validation loss depends on the voxel range of the fitting and validation subtomos, as subtomos are not normalised for the loss calculation. If the validation subtomos happen to be sampled from regions of the tomogram that contain less density or sample than the rest of the tomogram, the validation loss will be smaller than the fitting loss. By the same argument, it is also possible for the validation loss to be higher or similar to the fitting loss, as is the case in the tutorial (see `tutorial/Step2:ModelFitting.ipynb`). In any case, you are right that the shape of the validation loss can be used to identify overfitting.

However, I am also surprised that you _consistently_ observe lower validation losses, as the relative order of fitting and validation loss should be random in each run. To see if this is indeed the case, you could start two runs on the same data but change the random seed in the config file, which should lead to a different fitting/validation split.
Finally, as always, your observation could be caused by a bug or mistake on my part. I just checked the validation routine and it seemed fine to me, but I will take another, closer look and will let you know if I notice anything funny.

Best,
Simon

**Comment by rdrighetto** [comment]
Hi Simon,

Good catch! Indeed I've been running DDW always with 42 as the random seed. When changing that to 229, indeed the relative scaling of the validation and fitting losses changes:
![metrics](https://github.com/MLI-lab/DeepDeWedge/assets/7696255/123ac278-3bc2-4fa4-8501-21614c730825)
![metrics_rndtest](https://github.com/MLI-lab/DeepDeWedge/assets/7696255/fc9cf430-e14e-4d90-9a8d-2bc4fa9367a7)
The two runs above are identical except for the random seed, and one was launched much later than the other so it is lagging behind in the epochs, but we can already see what we wanted to see 🙂

Thanks for the quick reply!

**Comment by SimWdm** [comment]
Great news! Thanks for trying with different seeds. 👍
